AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSDataOperationProxy.m

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+#import "MSDataOperationProxy.h"
+#import "MSDataStorageConstants.h"
+#import "MSDataStore.h"
+#import "MSDataStoreErrors.h"
+#import "MSDataStoreInternal.h"
+#import "MSDocumentUtils.h"
+#import "MSDocumentWrapperInternal.h"
+#import "MSLogger.h"
+#import "MSTokenExchange.h"
+#import "MSTokensResponse.h"
+
+@implementation MSDataOperationProxy : NSObject
+
+- (instancetype)initWithDocumentStore:(id<MSDocumentStore>)documentStore {
+  self = [super init];
+  if (self) {
+    _documentStore = documentStore;
+    _reachability = [MS_Reachability reachabilityForInternetConnection];
+  }
+  return self;
+}
+
+- (void)performOperation:(NSString *_Nullable)operation
+              documentId:(NSString *)documentId
+            documentType:(Class)documentType
+                document:(id<MSSerializableDocument> _Nullable)document
+             baseOptions:(MSBaseOptions *_Nullable)baseOptions
+        cachedTokenBlock:(void (^)(MSCachedTokenCompletionHandler))cachedTokenBlock
+     remoteDocumentBlock:(void (^)(MSDocumentWrapperCompletionHandler))remoteDocumentBlock
+       completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+
+  // Get effective device time to live.
+  NSInteger deviceTimeToLive = baseOptions ? baseOptions.deviceTimeToLive : kMSDataStoreTimeToLiveDefault;
```

Comment:
nit: you can simplify since accessing a property on a nil object will also return nil
```suggestion
  NSInteger deviceTimeToLive = baseOptions.deviceTimeToLive ?: kMSDataStoreTimeToLiveDefault;
```